### PR TITLE
My orders

### PR DESCRIPTION
**Pull request by ileenshaar**

### CODE REVIEW

File: src/components/accordion-component/AccordionComponent.jsx [hunk 1]
```diff
@@ -12,9 +12,18 @@ export function AccordionComponent({
...
+    <Accordion
+      defaultExpanded={isExpanded}
+      style={{ margin: 0 }}
+      sx={{ boxShadow: 0, "&::before": { display: "none" }, margin: "0" }}
+      {...props}
+    >
       <AccordionSummary
-        expandIcon={<ExpandMoreIcon sx={{ color: "var(--dark)", fontSize: "2rem" }} />}
+        expandIcon={
+          isExpandIcon && (
```
Comment (mahmoud-alragabi) [inline src/components/accordion-component/AccordionComponent.jsx]: rename to ```showExpandIcon```

File: src/components/fetch-url/FetchUrl.js
Comment (mahmoud-alragabi) [inline src/components/fetch-url/FetchUrl.js:1]: great module, this should be in modules, and should have a better name than fetch URL i's too vague

File: src/components/profile-sidebar/ProfileSidebar.jsx [hunk 2]
```diff
@@ -1,54 +1,47 @@
...
-    id: `vertical-tab-${index}`,
-    'aria-controls': `vertical-tabpanel-${index}`,
-  };
-}
-
-export function ProfileSidebar({ SidebarOptions, titleSetter }) {
+export function ProfileSidebar({
+  SidebarOptions,
+  titleSetter,
+  setselectedOrderId,
```
Comment (mahmoud-alragabi) [inline src/components/profile-sidebar/ProfileSidebar.jsx]: rename to ```setSelectedOrderId```

File: src/components/profile-sidebar/ProfileSidebar.jsx [hunk 3]
```diff
@@ -1,54 +1,47 @@
...
+  const location = useLocation();
+  const navigate = useNavigate();
 
   const handleChange = (event, newValue) => {
     setValue(newValue);
     titleSetter(newValue);
+    setselectedOrderId("");
+    const newPath =
+      newValue !== 0
+        ? SidebarOptions[newValue].label.toLowerCase().split(" ").join("-")
```
Comment (mahmoud-alragabi) [inline src/components/profile-sidebar/ProfileSidebar.jsx]: should have this whole operation in a clearly named variable, or be it's own function

File: src/components/profile-sidebar/ProfileSidebar.jsx [hunk 4]
```diff
@@ -1,54 +1,47 @@
...
+        : "";
+    navigate(`/user-profile/${newPath}`);
   };
 
+  useEffect(() => {
+    const pathSegments = location.pathname.split("/");
+    const newPath = pathSegments[pathSegments.length - 1];
+
+    const selectedIndex = SidebarOptions.findIndex((option) =>
+      option.label.toLowerCase().split(" ").join("-").includes(newPath)
```
Comment (mahmoud-alragabi) [inline src/components/profile-sidebar/ProfileSidebar.jsx]: should be in a function

File: src/components/tab-navigation/TabNavigation.jsx [hunk 5]
```diff
@@ -22,33 +22,33 @@ export function TabNavigation({
...
-    </div>
+    <Tabs
+      onChange={handleChange}
+      defaultValue={0}
+      aria-label="Tabs Navigations"
+      selectionFollowsFocus
+    >
+      <TabsList style={{ padding: tabsPadding }} className={styles.tabList}>
+        {titles.map((title, index) => (
+          <Tab
```
Comment (mahmoud-alragabi) [inline src/components/tab-navigation/TabNavigation.jsx:43]: needs a key

File: src/pages/my-cart/MyCart.jsx [hunk 6]
```diff
@@ -55,8 +55,12 @@ export const MyCart = () => {
...
           <OrderInfo title={"Order Summary"} orderInfo={orderDetails} />
           <Box className={styles.operations}>
-            <CustomButton href='/checkout' label={"Place Order"} />
-            <CustomButton href='/' label={"Continue Shopping"} variant="outlined" />
+            <CustomButton href="/checkout" label={"Place Order"} />
+            <CustomButton
+              href="/"
+              label={"Continue Shopping"}
+              variant="outlined"
+            />
```
Comment (mahmoud-alragabi) [inline src/pages/my-cart/MyCart.jsx]: don't use href use router links

File: src/pages/user-profile/UserProfile.jsx [hunk 7]
```diff
@@ -1,57 +1,155 @@
...
+    },
+    {
+      label: "My Saved Cards",
+    },
+  ];
+
+  useEffect(() => {
+    const fetchDataAsync = async () => {
+      try {
+        const fetchedData = await fetchData("orders");
```
Comment (mahmoud-alragabi) [inline src/pages/user-profile/UserProfile.jsx]: i think ```fetchedOrders``` would be a better name

File: src/pages/user-profile/UserProfile.jsx [hunk 8]
```diff
@@ -1,57 +1,155 @@
...
+            />
+            <Route index element={<UserInformation />} />
+          </Routes>
         </Box>
-    )
-}
+      </Box>
+    </Box>
+  );
+};
```
Comment (mahmoud-alragabi) [inline src/pages/user-profile/UserProfile.jsx]: can definitely be broken down to smaller components

File: src/pages/user-profile/components/items-ordered/ItemsOrdered.jsx [hunk 9]
```diff
@@ -1,27 +1,58 @@
...
-  paymentDetails,
-  addressDetails,
-}) {
   return (
     <Box sx={{ display: "flex", flexDirection: "column", gap: "40px" }}>
       <TabNavigation
         titles={["Items Ordered", "Invoices", "Order Shipment"]}
         tabPadding="12px 16px"
         tabsPadding="12px 18px"
+        handleTabClick={() => console.log("tab clicked")}
```
Comment (mahmoud-alragabi) [inline src/pages/user-profile/components/items-ordered/ItemsOrdered.jsx]: remove the ```console.log``` it's a bad practice

File: src/pages/user-profile/components/my-orders/components/orders-table/OrdersTable.jsx [hunk 10]
```diff
@@ -2,7 +2,14 @@ import { Box, Checkbox, Typography } from "@mui/material";
 import { OrdersGrid } from "../orders-grid";
 import { RightChevron } from "../../../../../../components/icons";
 
-export function OrdersTable({ ordertablegrid, order }) {
+export function OrdersTable({ ordertablegrid, order, onOrderClick, total }) {
```
Comment (mahmoud-alragabi) [inline src/pages/user-profile/components/my-orders/components/orders-table/OrdersTable.jsx]: camel case ```orderTableGrid```

**Review by mahmoud-alragabi — CHANGES_REQUESTED** [review]

**Review by mahmoud-alragabi — APPROVED** [review]

**Review by ferassmohana — APPROVED** [review]

**Pull request merged**